### DISCUSSION TITLE
I need to understand how the algorithm works

### DISCUSSION THREAD

**Discussion by athtareq** [lululxvi/deepxde#672] [Q&A] [answered]
Hello, I'm using DeepXDE (great job btw) for a project and I just need you to explain an example for me, you know just what each line does and all, since the .py examples are kind of different to the procedure that's listed in the official paper, and also how does the output work ? I know this may seem pretty elementary but I need some clarification, thanks !  

` 

    def boundary_l(x, on_boundary):
        return on_boundary and np.isclose(x[0], -1)

    def boundary_r(x, on_boundary):
        return on_boundary and np.isclose(x[0], 1)

    def func(x):
        return (x + 1) ** 2

    geom = dde.geometry.Interval(-1, 1)
    bc_l = dde.DirichletBC(geom, func, boundary_l)
    bc_r = dde.NeumannBC(geom, lambda X: 2 * (X + 1), boundary_r)
    data = dde.data.PDE(geom, 1, pde, [bc_l, bc_r], 16, 2, func=func, num_test=100)

    layer_size = [1] + [50] * 3 + [1]
    activation = "tanh"
    initializer = "Glorot uniform"
    net = dde.maps.FNN(layer_size, activation, initializer)

    model = dde.Model(data, net)
    model.compile("adam", lr=0.001, metrics=["l2 relative error"])
    losshistory, train_state = model.train(epochs=10000)

    dde.saveplot(losshistory, train_state, issave=True, isplot=True)

 `

**Comment by lululxvi** [comment, marked as answer]
Please check the slides in https://github.com/lululxvi/deepxde: https://lululxvi.github.io/files/talks/2020PIML.pdf

**Comment by athtareq** [comment]
Exactly what I wanted, thank you !
